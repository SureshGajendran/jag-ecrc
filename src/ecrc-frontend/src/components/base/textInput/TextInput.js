import React from "react";
import PropTypes from "prop-types";
import "./TextInput.css";

export const TextInput = ({
  textInput: { label, id, note, textInputStyle, value, isRequired },
  onChange
}) => {
  let asterisk = "";
  if (isRequired) {
    asterisk = (
      <span id="asterisk" className="mandatory">
        *
      </span>
    );
  }

  const labelPart = (
    <div className="label">
      <label className="textinput_label" htmlFor={id}>
        {label}
      </label>
      {asterisk}
      <span className="note">&nbsp;{note}</span>
    </div>
  );

  let textStyle = "textinput_editable_white";
  const validStyles = [
    "textinput_editable_gray",
    "textinput_non_editable_gray",
    "textinput_editable_white"
  ];

  if (validStyles.includes(textInputStyle)) {
    textStyle = textInputStyle;
  }

  if (textStyle === "textinput_non_editable_gray")
    return (
      <div>
        {labelPart}
        <input
          className={`${textStyle}`}
          type="text"
          id={id}
          defaultValue={value}
          readOnly
          onChange={onChange}
        />
      </div>
    );

  return (
    <div>
      {labelPart}
      <input
        className={`${textStyle}`}
        type="text"
        id={id}
        defaultValue={value}
        onChange={event => onChange(event.target.value)}
      />
    </div>
  );
};

TextInput.propTypes = {
  textInput: PropTypes.shape({
    label: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    note: PropTypes.string,
    textInputStyle: PropTypes.string,
    value: PropTypes.string,
    isRequired: PropTypes.bool
  }).isRequired,
  onChange: PropTypes.func
};

TextInput.defaultProps = {
  note: "",
  onChange: () => {}
};

export default TextInput;
