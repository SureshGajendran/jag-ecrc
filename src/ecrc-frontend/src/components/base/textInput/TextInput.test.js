import React from "react";
import { create } from "react-test-renderer";
import { TextInput } from "./TextInput";
import { shallow } from "enzyme";

describe("TextInput Component", () => {
  const textInput = {
    label: "button label",
    id: "textInputId",
    value: "init value"
  };

  test("Matches the snapshot", () => {
    const textInputBox = create(<TextInput textInput={textInput} />);
    expect(textInputBox.toJSON()).toMatchSnapshot();
  });

  test("Renders the textInput when the style is not editable", () => {
    const textInputBox = shallow(
      <TextInput
        textInput={{
          ...textInput,
          textInputStyle: "textinput_non_editable_gray"
        }}
      />
    );
    expect(textInputBox.find("div input").prop("readOnly")).toBe(true);
  });

  test("Renders the textInput correctly when isRequired is set", () => {
    const textInputBox = shallow(
      <TextInput
        textInput={{
          ...textInput,
          isRequired: true
        }}
      />
    );
    expect(textInputBox.exists("#redStar")).toEqual(true);
  });
});
