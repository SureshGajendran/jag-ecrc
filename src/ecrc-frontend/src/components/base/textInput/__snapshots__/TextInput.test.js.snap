// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component Matches the snapshot 1`] = `
<div>
  <div>
    <label
      className="textinput_label"
      htmlFor="textInputId"
    >
      button label
    </label>
    
  </div>
  <input
    className="textinput_editable_white"
    defaultValue="init value"
    id="textInputId"
    onChange={[Function]}
    type="text"
  />
</div>
`;
